### ISSUE TITLE
Very slow infer on saved ckpt

### ISSUE THREAD

**Issue by BrightHai** [om-ai-lab/VLM-R1#79]
trained on 1 x H20 96G，it takes very long time when inferring image with the saved model, normal with the origin model qwen-2.5-vl-3B

**Infer code**
import torch
from qwen_vl_utils import process_vision_info
from modelscope import AutoTokenizer
from transformers import (
    Qwen2_5_VLForConditionalGeneration,
    AutoProcessor,
)
import json
import os

prompt = """
描述图像
===回答方式===
<think></think>
<answer></answer>
"""

prompt = "描述图像 在<think> </think>标签中输出思考过程，在<answer> </answer>标签中输出最终答案"

resized_height = 1170
resized_width = 460
MAX_LENGTH = 8192

merge_model_path = 'autodl-tmp/r1-output/checkpoint-10'
val_peft_model = Qwen2_5_VLForConditionalGeneration.from_pretrained(
    merge_model_path, device_map="auto", torch_dtype=torch.bfloat16, 
    trust_remote_code=True,attn_implementation="flash_attention_2")
tokenizer = AutoTokenizer.from_pretrained(merge_model_path, use_fast=False, trust_remote_code=True)
processor = AutoProcessor.from_pretrained(merge_model_path)

def predict(messages, model):
    # 准备推理
    text = processor.apply_chat_template(
        messages, tokenize=False, add_generation_prompt=True
    )
    image_inputs, video_inputs = process_vision_info(messages)
    inputs = processor(
        text=[text],
        images=image_inputs,
        videos=video_inputs,
        padding=True,
        return_tensors="pt",
    )
    inputs = inputs.to("cuda")

    # 生成输出
    generated_ids = model.generate(**inputs, max_new_tokens=MAX_LENGTH, temperature=0.1)
    generated_ids_trimmed = [
        out_ids[len(in_ids) :] for in_ids, out_ids in zip(inputs.input_ids, generated_ids)
    ]
    output_text = processor.batch_decode(
        generated_ids_trimmed, skip_special_tokens=True, clean_up_tokenization_spaces=False
    )

    return output_text[0]

image_file_path = 'label_image/1672_1.png'
messages = [{
    "role": "user",
    "content": [
        {
            "type": "image",
            "image": image_file_path,
            "resized_height": resized_height,
            "resized_width": resized_width
        },
        {
            "type": "text",
            "text": prompt
        }
    ]}]

response = predict(messages, val_peft_model)
print(response)

**train code**
python src/open-r1-multimodal/src/open_r1/grpo_jsonl.py \
    --output_dir /root/autodl-tmp/r1-output \
    --model_name_or_path /root/autodl-tmp/qwen2.5_vl_200 \
    --data_file_paths /root/labels_r1.jsonl \
    --image_folders /root/label_image/ \
    --max_prompt_length 1024 \
    --max_completion_length 1024 \
    --num_generations 4 \
    --per_device_train_batch_size 1 \
    --gradient_accumulation_steps 2 \
    --logging_steps 1 \
    --bf16 \
    --torch_dtype bfloat16 \
    --data_seed 42 \
    --report_to wandb \
    --gradient_checkpointing true \
    --attn_implementation flash_attention_2 \
    --num_train_epochs 2 \
    --run_name ui-detect \
    --dataset_name ui-detect \
    --save_steps 100 \
    --save_only_model true

**Comment by BrightHai** [comment]
set use_cache=true
